test/development/acceptance-app/undefined-default-export.test.ts

```diff
@@ -23,6 +23,25 @@ describe('Undefined default export', () => {
     )
   })
 
+  it('should error if layout component does not have default export', async () => {
+    await using sandbox = await createSandbox(
+      next,
+      new Map([
+        ['app/(group)/specific-path/server/layout.js', 'export const a = 123'],
+        [
+          'app/(group)/specific-path/server/page.js',
+          'export default function Page() { return <div>Hello</div> }',
+        ],
+      ]),
+      '/specific-path/server'
+    )
+    const { session } = sandbox
+    await session.assertHasRedbox()
+    expect(await session.getRedboxDescription()).toInclude(
+      'The default export is not a React Component in "/specific-path/server/layout"'
```

Comment:
That's not a valid URL though. I think the real bug is that we're not actually passing around the path to that component and rather some internal identifier that is neither URL nor path.